code/datums/martial/monk.dm

```diff
@@ -0,0 +1,455 @@
+	var/ki_level = 0
+	var/cleave_level = 0
+	var/circle_kick = FALSE
+	var/diamond_body = FALSE
+	var/purity_of_body = FALSE
+	var/using_flurry = TRUE
+	var/is_disciple = FALSE
+
+	var/list/available_actions = list()
+	var/datum/action/monk_rest/monk_rest = new/datum/action/monk_rest()
+	var/datum/action/make_disciple/make_disciple = new/datum/action/make_disciple()
+	var/datum/action/flurry_toggle/flurry_toggle = new/datum/action/flurry_toggle()
+	var/datum/action/monk/stunning_fist/stunning_fist = new/datum/action/monk/stunning_fist()
+	var/datum/action/monk/quivering_palm/quivering_palm = new/datum/action/monk/quivering_palm()
+	var/datum/action/monk/wholeness_of_body/wholeness_of_body = new/datum/action/monk/wholeness_of_body()
+
+/datum/martial_art/monk/disciple
+	name = "Disciple Monk"
+	level_cap = 20
+	is_disciple = TRUE
+
+
+/datum/martial_art/monk/teach(mob/living/carbon/human/H,make_temporary=0)
+	if(..())
+		if(H.client && H.client.prefs.custom_names["human"])
+			H.fully_replace_character_name(H.real_name, H.client.prefs.custom_names["human"])
+		else
+			H.fully_replace_character_name(H.real_name, random_unique_name(H.gender))
+		H.set_species(/datum/species/human)
+		if(H.dna.features["tail_human"] == "Cat")
+			H.dna.features["tail_human"] -= "Cat"
+			H.dna.species.mutant_bodyparts -= "tail_human"
+		if(H.dna.features["ears"] == "Cat")
+			H.dna.features["ears"] -= "Cat"
+			H.dna.species.mutant_bodyparts -= "ears"
+		H.dna.species.no_equip += ITEM_SLOT_HEAD
+		monk_rest.Grant(H)
+		flurry_toggle.Grant(H)
+		flurry_toggle.martial = src
+		ADD_TRAIT(H, TRAIT_NOGUNS, MONK_TRAIT)
+		to_chat(H, "<span class = 'userdanger'>You know the ancient Monk techniques!</span>")
+		to_chat(H, "<span class = 'danger'>This technique becomes more powerful the more you use it.</span>")
+		to_chat(H, "<span class = 'danger'>Your power level will increase from 1 to 20, gaining new abilities and growing stronger the more you train.</span>")
+		to_chat(H, "<span class = 'danger'>Your abilities have a use limit, before you need to rest to regain your strength.</span>")
+		to_chat(H, "<span class = 'danger'>You can't gain experience on yourself, dead humans, non humans, or braindead humans.</span>")
+		START_PROCESSING(SSfastprocess, src)
+		H.hairstyle = "Bald"
+		H.update_body()
+		H.update_hair()
+
+/datum/martial_art/monk/proc/flurry_penalty()
+	if(using_flurry)
+		return flurry_of_blows_penalty
+	return 0
+
+/datum/martial_art/monk/on_remove(mob/living/carbon/human/H)
+	to_chat(H, "<span class = 'userdanger'>You forget the ways of a Monk...</span>")
+	stunning_fist.Remove(H)
+	quivering_palm.Remove(H)
+	flurry_toggle.Remove(H)
+	monk_rest.Remove(H)
+	wholeness_of_body.Remove(H)
+	make_disciple.Remove(H)
+	REMOVE_TRAIT(H, TRAIT_NOGUNS, MONK_TRAIT)
+	STOP_PROCESSING(SSfastprocess, src)
+
+/datum/martial_art/monk/process()
+	..()
+	if(diamond_body)
+		owner.reagents.remove_all_type(/datum/reagent/toxin, 30)

```

Comment:
Just give the TOXIMMUNE trait.

Reply:
I'll take care of it soon and add a anti virus trait the name I forgot of too.

Reply:
Toximmune didn't exist at the time of this coding, my apologies.